File: npm/src/serve/api_attention.js
'use strict'

const fs = require('fs')
const path = require('path')

function getAttention(cfg) {
  const attentionPath = path.join(cfg.roadmapDir, '.trackfw-attention.json')
  try {
    const payload = JSON.parse(fs.readFileSync(attentionPath, 'utf8'))
    return { ...payload, active: true }
  } catch (_) {
    return { active: false }
  }
}

function handleAttention(cfg, req, res) {
  res.writeHead(200, {
    'Content-Type': 'application/json',
    'Access-Control-Allow-Origin': '*'
  })
  res.end(JSON.stringify(getAttention(cfg)))
}

module.exports = { getAttention, handleAttention }

File: pypi/trackfw/serve/api_attention.py
"""Attention signal API used by the local dashboard."""

import json
import os


def get_attention(cfg):
    attention_path = os.path.join(
        cfg.get("roadmap_dir", "docs/roadmaps"),
        ".trackfw-attention.json",
    )
    try:
        with open(attention_path, encoding="utf-8") as stream:
            payload = json.load(stream)
    except (OSError, ValueError, TypeError):
        return {"active": False}

    payload["active"] = True
    return payload
